src/inc/RingBuffer.h

```diff
@@ -58,17 +57,31 @@ class CRingBuffer
         incIdx(m_iReadIdx);
         return tValue;
     }
-
+//    T getPostInc(float fOffset = 0.f) const
+//    {
+//        T tValue = get(fOffset);
+//        incIdx(m_iReadIdx);
+//        return tValue;
+//    }
     /*! return the value at the current read index
     \param fOffset: read at offset from read index
     \return float the value from the read index
     */
-    T get(float fOffset = 0) const
+    T get(float fOffset = 0.f) const
     {
-        assert(0); // TODO: implement offset
-        return m_ptBuff[m_iReadIdx];
-    }
+        int readIdx = (m_iReadIdx + floor(fOffset));
```

Comment:
I think adding a conditional for 0 offset for direct retrieval would cut processing time (avoid the mod)